base/compiler/abstractinterpretation.jl

```diff
@@ -862,8 +862,12 @@ function abstract_iteration(interp::AbstractInterpreter, @nospecialize(itft), @n
     # the precise (potentially const) state type
     statetype = widenconst(statetype)
     valtype = widenconst(valtype)
-    while valtype !== Any
+    if statetype === Bottom
```

Comment:
Is the issue here when skipping the loop above? Maybe we should directly widen stateordonet then? Otherwise we're redoing this call here which duplicates work that was already done above.

Reply:
Widening `stateordonet` may not be sufficient, we need to compute it from the widened `statetype` as input to `iterate`. However, I also dislike the duplicaton of work here. (Which, btw, we also did before as the loop started with computing a new `stateordonet`.) However, I think I have found an alternative that avoids the duplication at the cost of replacing a `<:` with `⊑`. Will push momentarily; let me know what you think.